langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/gemini/GeminiAiChatLanguageModelIT.java

```diff
@@ -0,0 +1,597 @@
+package dev.langchain4j.model.gemini;
+
+import com.google.gson.Gson;
+import dev.langchain4j.agent.tool.JsonSchemaProperty;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.AudioContent;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ImageContent;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.TextContent;
+import dev.langchain4j.data.message.TextFileContent;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.ResponseFormat;
+import dev.langchain4j.model.chat.request.json.JsonEnumSchema;
+import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
+import dev.langchain4j.model.chat.request.json.JsonSchema;
+import dev.langchain4j.model.chat.request.json.JsonSchemaElement;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.output.JsonSchemas;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.Utils.readBytes;
+import static dev.langchain4j.model.chat.request.ResponseFormatType.JSON;
+import static dev.langchain4j.model.chat.request.json.JsonStringSchema.JSON_STRING_SCHEMA;
+import static dev.langchain4j.model.gemini.GeminiAiChatLanguageModel.pythonCodeExecution;
+import static dev.langchain4j.model.gemini.GeminiHarmBlockThreshold.BLOCK_LOW_AND_ABOVE;
+import static dev.langchain4j.model.gemini.GeminiHarmCategory.HARM_CATEGORY_HARASSMENT;
+import static dev.langchain4j.model.gemini.GeminiHarmCategory.HARM_CATEGORY_HATE_SPEECH;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(RetryExtension.class)
```

Comment:
major: something is wrong with the test execution. even if the test assertion fails, junit says it is green

Reply:
I've removed my custom assertion, and used one from JUnit Pioneer (just a test dependency)
I'm using Pioneer's `@RetryingTest(n)` annotation now, which works at a method test level rather than class+method level.
It's for flaky tests which sometimes fail because Gemini replies a bit differently. That's the case for example for parallel too calling or safety settings tests, as Gemini doesn't 100% reliably behave the same way, once in a while.